apis/Google.Cloud.Diagnostics.Common/Google.Cloud.Diagnostics.Common.IntegrationTests/Trace/TraceHeaderPropagatingHandlerTest.cs

```diff
@@ -0,0 +1,144 @@
+
+            var trace = await TestTracingOutGoingRequest(fakeUri, spanName, shouldThrow: true);
+            var innerSpan = trace.Spans.Single(s => s.Name != spanName);
+            var label = innerSpan.Labels.Single();
+            Assert.Equal(TraceLabels.StackTrace, label.Key);
+            Assert.False(string.IsNullOrWhiteSpace(label.Value));
+        }
+
+        /// <summary>
+        /// Helper funtion to test tracing of out going requests.
+        /// </summary>
+        /// <param name="uri">The uri to call.</param>
+        /// <param name="rootSpanName">The name for the root span.</param>
+        /// <param name="shouldThrow">True if the request to the uri should throw.</param>
+        /// <returns>A trace with two spans.  A root span and a child span tracing the outgoing uri
+        ///     request.</returns>
+        private async Task<TraceProto> TestTracingOutGoingRequest(
+            string uri, string rootSpanName, bool shouldThrow)
+        {
+            var startTime = Timestamp.FromDateTime(DateTime.UtcNow);
+            await TraceOutGoingRequest(CreateTracer(), rootSpanName, uri, shouldThrow);
+            var trace = _polling.GetTrace(rootSpanName, startTime);
+            CheckTrace(trace, rootSpanName, uri);
+            return trace;
+        }
+
+        /// <summary>
+        /// Creates a <see cref="SimpleManagedTracer"/> with a <see cref="GrpcTraceConsumer"/>.
+        /// </summary>
+        private SimpleManagedTracer CreateTracer()
+        {
+            string traceId = _traceIdFactory.NextId();
+            var traceProto = new TraceProto { ProjectId = _projectId, TraceId = traceId };
+            var consumer = new GrpcTraceConsumer(TraceServiceClient.CreateAsync());
+            return SimpleManagedTracer.Create(consumer, traceProto, null);
+        }
+
+        /// <summary>
+        /// Creates a <see cref="TraceHeaderPropagatingHandler"/> and traces the sending of a
+        /// GET request to the given uri.  The trace is wrapped in a parent span.
+        /// </summary>
+        /// <param name="tracer">The tracer to trace the request with.</param>
+        /// <param name="rootSpanName">The name of the root span that will wrap the span
+        ///     that traces the request.</param>
+        /// <param name="uri">The uri to request.</param>
+        /// <param name="shouldThrow">True if the request will throw.</param>
+        private async Task TraceOutGoingRequest(
+            IManagedTracer tracer, string rootSpanName, string uri, bool shouldThrow)
+        {
+            tracer.StartSpan(rootSpanName);
+            var traceHeaderHandler = TraceHeaderPropagatingHandler.Create(tracer);
+            using (var httpClient = new HttpClient(traceHeaderHandler))
+            {
+                try
+                {
+                    await httpClient.GetAsync(uri);
+                    Assert.False(shouldThrow);
+                }
+                catch
```

Comment:
Do you mean to catch the thrown exception when `shouldThrow` is true? If so, you may want to add that to the param docs above. They currently make it sound like the `TraceOutGoingRequest` call will throw when it is true, but it will not.